iroh/src/node.rs

```diff
@@ -1441,31 +1441,25 @@ impl<D: BaoStore> RpcHandler<D> {
     ) -> impl Stream<Item = RpcResult<NodeConnectionsResponse>> + Send + 'static {
         // provide a little buffer so that we don't slow down the sender
         let (tx, rx) = flume::bounded(32);
+        let mut conn_infos = self.inner.endpoint.connection_infos();
+        conn_infos.sort_by_key(|n| n.public_key.to_string());
         self.rt().spawn_pinned(|| async move {
-            match self.inner.endpoint.connection_infos().await {
-                Ok(mut conn_infos) => {
-                    conn_infos.sort_by_key(|n| n.public_key.to_string());
-                    for conn_info in conn_infos {
-                        tx.send_async(Ok(NodeConnectionsResponse { conn_info }))
-                            .await
-                            .ok();
-                    }
-                }
-                Err(e) => {
-                    tx.send_async(Err(e.into())).await.ok();
-                }
+            for conn_info in conn_infos {
+                tx.send_async(Ok(NodeConnectionsResponse { conn_info }))
+                    .await
+                    .ok();
             }
         });
         rx.into_stream()
     }
 
-    async fn node_connection_info(
+    fn node_connection_info(
         self,
         req: NodeConnectionInfoRequest,
-    ) -> RpcResult<NodeConnectionInfoResponse> {
+    ) -> impl Future<Output = RpcResult<NodeConnectionInfoResponse>> {
```

Comment:
why the change from `async` to `impl Future`?

Reply:
Because otherwise clippy says `Warning: unused `async` for function with no await statements`. Also maybe it makes it obvious that there is no await. Can also add a clippy allow, if preferred? Don't mind.

Reply:
Changed it to return `std::future::Ready` directly.